File: Pythonleetcode/Leetcode24.py
import list
class Solution(object):

    def swapPairs(self, head):

        if head == None:
            return head

        cur = ListNode(0)

        cur.next = head

        first = cur

        while cur.next and cur.next.next:
            n1 = cur.next

            n2 = n1.next

            nxt = n2.next

            n1.next = nxt

            n2.next = n1

            cur.next = n2

            cur = n1

        return first.next


print(Solution().swapPairs([1,2,3,4]))

File: CSharpPractice/LeetcodePractice/Leetcode24.cs
public class Solution24 {
    public ListNode SwapPairs(ListNode head) 
    {
        if (head==null)
        {
            return null;
        }
        else
        {
            if (head.next==null)
            {
                return head;
            }
        }
        ListNode slow = head;
        ListNode fast = slow.next;
        while (slow!=null)
        {
            if (fast!=null)
            {
                int temp = 0;
                temp=slow.val;
                slow.val = fast.val;
                fast.val=temp;
                slow = fast.next;
                if (slow!=null)
                {
                    fast = slow.next;
                }
                else
                {
                    break;
                }
            }
            else
            {
                break;
            }
        }
        return head;
    }
}